client/main.go

```diff
@@ -465,7 +465,14 @@ func main() {
 					jitterHist.RecordValue(jitter)
 					globalJitterHist.RecordValue(jitter)
 				}
-				if *totalRequests > 0 && totalCount >= *totalRequests {
+				if *totalRequests > 0 && totalCount + 1 >= *totalRequests {
+					if resp.err != nil {
+						bad++
+						totalBad++
+					} else {
+						good++
+						totalGood++
+					}
```

Comment:
Incrementing the `bad` and `good` counters here seems like we're double counting this request, since the same counters are incremented on lines 440-445.

I think the issue is that the client uses multiple channels to control requests that are being sent. Each concurrent worker sends requests continuously, and writes the responses to the shared `received` channel. In addition, each worker has a separate `shutdown` channel, which is used to notify the worker to stop sending requests. Once `totalRequests` have been sent, this process writes to the `shutdown` channels, but that does not stop requests that are already in flight. Hence, when the client is invoked with `-totalRequests=10 -concurrency=1`, 11 requests are sent. Likewise if it is invoked with `-totalRequests=10 -concurrency=2`, then 12 requests are sent, and so on.

I've tried to come up with another way to fix this issue but haven't arrived at anything specific. By the time the shutdown message is sent, the workers have already fired the next request. Changing the threshold for shutdown to `totalRequests + 1`, really only works in the case that the test is running with `concurrency=1`. We could change the threshold to `totalRequests + concurrency`, but I can imagine cases where that would result in fewer requests than desired being sent. I think the current behavior of just not recording the responses that are received after the shutdown message is sent is probably the most correct.